openedx/core/djangoapps/profile_images/views.py

```diff
@@ -81,11 +95,19 @@ def post(self, request, username):
                         status=status.HTTP_400_BAD_REQUEST,
                     )
 
+                # come up with a unique version for the uploaded pic
+                new_version = _make_image_version()
+
                 # generate profile pic and thumbnails and store them
-                create_profile_images(uploaded_file, get_profile_image_names(username))
+                create_profile_images(uploaded_file, get_profile_image_names(username, new_version))
+
+                # if there was previously a different image, delete it
+                old_version = get_profile_image_version(username)
+                if old_version is not None:
+                    remove_profile_images(get_profile_image_names(username, old_version))
 
                 # update the user account to reflect that a profile image is available.
-                set_has_profile_image(username, True)
+                set_profile_image_version(username, new_version)
```

Comment:
CloudFront supports querystrings, though it's suggested that it could impact performance.